bdb/tran.c

```diff
@@ -2787,63 +2785,6 @@ void bdb_unregister_modsnap(bdb_state_type *bdb_state, void * registration)
     free(outstanding_modsnap);
 }
 
-int bdb_get_modsnap_start_state(bdb_state_type *bdb_state,
-                        int is_ha_retry,
-                        int snapshot_epoch,
-                        unsigned int *modsnap_start_lsn_file,
-                        unsigned int *modsnap_start_lsn_offset,
-                        unsigned int *last_checkpoint_lsn_file,
-                        unsigned int *last_checkpoint_lsn_offset)
-{
-    DB_ENV *dbenv;
-    DB_LSN modsnap_start_lsn;
-    DB_LSN last_checkpoint_lsn;
-    DB_TXN_COMMIT_MAP *txmap;
-    int rc;
-
-    rc = 0;
-    dbenv = bdb_state->dbenv;
-    txmap = dbenv->txmap;
-
-    if (is_ha_retry) {
-        modsnap_start_lsn.file = *modsnap_start_lsn_file;
-        modsnap_start_lsn.offset = *modsnap_start_lsn_offset;
-
-        bdb_checkpoint_list_get_ckplsn_before_lsn(modsnap_start_lsn, &last_checkpoint_lsn);
-    } else if (snapshot_epoch) {
-        // This LSN is not necessarily a commit LSN but this is okay --
-        // it can still serve as our start point.
-        bdb_get_lsn_context_from_timestamp(bdb_state, snapshot_epoch, &modsnap_start_lsn, 0, &rc); 
-        if (rc != 0) {
-            return rc;
-        }
-
-        bdb_checkpoint_list_get_ckplsn_before_lsn(modsnap_start_lsn, &last_checkpoint_lsn);
-    } else {
-        Pthread_mutex_lock(&txmap->txmap_mutexp);
-        if ((rc = __txn_commit_map_get_highest_checkpoint_lsn(dbenv, &last_checkpoint_lsn, 0)) != 0) {
```

Comment:
Removed `__txn_commit_map_get_highest_checkpoint_lsn` because we can just use `bdb_checkpoint_list_get_ckplsn_before_lsn`